### PR TITLE
add support for formats and other missing properties

### PR DESCRIPTION
**Pull request by albertshau**
Added support for formats, except restricting the UI and docs to leave out avro, parquet, and orc due to the lack of file seek in FTP. Also added/implemented several standard config properties supported by the other file based sources, like delimiter, whether to allow empty input, schema, etc.

### CODE REVIEW

File: docs/FTP-batchsource.md [hunk 1]
```diff
@@ -17,28 +17,41 @@ Properties
...
-Path is expected to be of the form prefix://username:password@hostname:port/path (Macro-enabled)
+Path is expected to be of the form prefix://username:password@hostname:port/path
 
-**Regex Path Filter:** Regex to filter out files in the path. It accepts regular expression which is applied to the complete
-path and returns the list of files that match the specified pattern. (Macro-enabled)
+**Format:** Format of the data to read.
+The format must be one of 'blob', 'csv', 'delimited', 'json', 'text', 'tsv', or the
+name of any format plugin that you have deployed to your environment. Note that FTP does
+not support seeking in a file, so formats like avro and parquet cannot be used.
+If the format is a macro, only the pre-packaged formats can be used.
```
Comment (CuriousVini) [inline docs/FTP-batchsource.md]: shall we say actual list of formats instead of `pre-packaged`?

File: widgets/FTP-batchsource.json [hunk 2]
```diff
@@ -18,6 +18,84 @@
...
+          "widget-type": "textbox",
+          "label": "Delimiter",
+          "name": "delimiter",
+          "widget-attributes": {
+            "placeholder": "Delimiter if the format is 'delimited'"
+          }
+        },
+        {
+          "widget-type": "toggle",
+          "name": "enableQuotedValues",
```
Comment (CuriousVini) [inline widgets/FTP-batchsource.json:70]: is delimiter and enableQuotedValues inherited in File source config?

**Review by CuriousVini — APPROVED** [review]
minor questions. LGTM

File: widgets/FTP-batchsource.json (hunk 2 above)
Comment (albertshau) [reply widgets/FTP-batchsource.json:70]: these aren't actually part of the file source config class, but the AbstractFileSource will pass anything that is set in the PluginConfig will down to the format plugins, which do use these values.

Comment (CuriousVini) [reply widgets/FTP-batchsource.json:70]: I see, sounds good, thanks!

File: docs/FTP-batchsource.md [hunk 3]
```diff
@@ -17,28 +17,41 @@ Properties
...
-needed for the distributed file system. (Macro-enabled)
+**Get Schema:** Auto-detects schema from file. Supported formats are: csv, delimited, tsv, blob and text.
 
-**Allow Empty Input:** Identify if path needs to be ignored or not, for case when directory or file does not
-exists. If set to true it will treat the not present folder as 0 input and log a warning. Default is false.
+Blob - is set by default as field named 'body' of type bytes.
+
+Text - is set by default as two fields: 'body' of type bytes and 'offset' of type 'long'.
+
+JSON - is not supported, user has to manually provide the output schema.
```
Comment (robinrielley) [inline docs/FTP-batchsource.md]: Can this be two sentences?  JSON - is not supported. You must manually provide the output schema.

File: docs/FTP-batchsource.md [hunk 4]
```diff
@@ -17,28 +17,41 @@ Properties
...
+
+**Delimiter:** Delimiter to use when the format is 'delimited'. This will be ignored for other formats.
 
+**Use First Row as Header:** Whether to use the first line of each file as the column headers. Supported formats are 'text', 'csv', 'tsv', 'delimited'.
```
Comment (robinrielley) [inline docs/FTP-batchsource.md]: Can we add "and" before ", and 'delimited'.

**Pull request merged**